### ISSUE TITLE
Delete object call using boto3 API on a versioned bucket deletes the DELETED marker

### ISSUE THREAD

**Issue by mannyd28** [minio/minio#16224]
I am using the boto3 API in python and when calling the delete_object function on an object in a versioned bucket, the wrong version is getting deleted (that is, the "delete marker" is getting deleted instead of adding a new delete marker).

This was not happening in version RELEASE.2022-03-22T02-05-10Z
This started happening after we upgraded to version RELEASE.2022-11-26T22-43-32Z
This is still happening in version RELEASE.2022-12-07T00-56-37Z 

## Expected Behavior
put_object  # v1 is created
delete_obejct  # v2 delete marker is created
put_object  # v3 is created 
delete_object  # v4 delete marker is created

## Current Behavior
put_object  # v1 is created
delete_obejct  # v2 delete marker is created
put_object  # v3 is created 
delete_object  # v2 delete marker is **deleted** and v3 becomes v2

## Possible Solution
n/a

## Steps to Reproduce (for bugs)
```python
import boto3
import time
from botocore.client import Config

ENDPOINT="http://localhost:9000"
ACCESS_KEY="minioadmin"
SECRET_KEY="minioadmin"
BUCKET_NAME="my-bucket"

filename = 'hello-world.txt'
file = b'hello world'

client = boto3.client('s3', endpoint_url=ENDPOINT, aws_access_key_id=ACCESS_KEY, aws_secret_access_key=SECRET_KEY, config=Config(signature_version='s3v4'), region_name='us-east-1')
client.create_bucket(Bucket=BUCKET_NAME, CreateBucketConfiguration={'LocationConstraint': 'us-east-1'})
client.put_bucket_versioning(Bucket=BUCKET_NAME, VersioningConfiguration={'MFADelete': 'Disabled', 'Status': 'Enabled'})

client.put_object(Body=file, Bucket=BUCKET_NAME, Key=filename, ContentType='text/plain')
objects = client.list_objects_v2(Bucket=BUCKET_NAME).get("Contents")
assert len(objects) == 1
assert objects[0]['Key'] == filename

client.delete_object(Bucket=BUCKET_NAME, Key=filename)
objects = client.list_objects_v2(Bucket=BUCKET_NAME).get("Contents")
assert objects is None

client.put_object(Body=file, Bucket=BUCKET_NAME, Key=filename, ContentType='text/plain')
objects = client.list_objects_v2(Bucket=BUCKET_NAME).get("Contents")
assert len(objects) == 1
assert objects[0]['Key'] == filename

# The following call will actually delete the DELETED marker created above
client.delete_object(Bucket=BUCKET_NAME, Key=filename)
objects = client.list_objects_v2(Bucket=BUCKET_NAME).get("Contents")
assert objects is None  # <- This assertion fails
```

## Context
Our application's automated integration tests are all failing due to this issue. 
Users must delete objects twice before they actually delete.

## Regression
n/a

## Your Environment

* Version used (`minio --version`): RELEASE.2022-12-07T00-56-37Z 
* Server setup and configuration: Docker container with 4-node distributed deployment
* Operating System and version (`uname -a`): RedHat UBI 8.7

**Comment by mannyd28** [comment]
### Verbose logs
[minio.log](https://github.com/minio/minio/files/10212652/minio.log)

### Before calling delete_object second time:
<img width="1167" alt="1-before-second-delete" src="https://user-images.githubusercontent.com/29721216/207176270-44cafad1-0ab9-49ec-a633-904bcc8f6c36.png">

### After calling delete_object the second time:
<img width="1167" alt="2-after-second-delete" src="https://user-images.githubusercontent.com/29721216/207176267-2e77d477-62a6-4695-b4f1-c58027c70243.png">

**Comment by poornas** [comment]
this is fixed in master - https://github.com/minio/minio/pull/16195

**Comment by harshavardhana** [comment]
> this is fixed in master - #16195

This is already fixed pending a release.
